adev/shared-docs/pipeline/navigation/types.ts

```diff
@@ -0,0 +1,28 @@
+// /*!
+//  * @license
+//  * Copyright Google LLC All Rights Reserved.
+//  *
+//  * Use of this source code is governed by an MIT-style license that can be
+//  * found in the LICENSE file at https://angular.dev/license
+//  */
+
+/**
+ * `NavigationItem` generation strategy
+ */
+export type NavigationItemGenerationStrategy = {
+  /**
+   * App route path prefix.
+   */
+  pathPrefix: string;
+  /**
+   * Content path where the source files are kept.
+   */
+  contentPath: string;
+  /**
+   * Page/route label generator function.
+   */
+  labelGeneratorFn: (fileName: string, firstLine: string) => string;
+};
+
+// All strategies names
```

Comment:
This should be a block comment for intellisense to use it.
```suggestion
/** Strategy for navigation item generation. */
```